core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -714,16 +744,37 @@ public void month() {
     assertEquals(integerValue(8), eval(expression));
   }
 
+  @Test
+  public void testMonthOfYearWithTimeType() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+    FunctionExpression expression = DSL.month_of_year(
+        functionProperties,
+        DSL.literal(new ExprTimeValue("12:23:34")));
+
+    assertEquals(INTEGER, eval(expression).type());
+    assertEquals(LocalDate.now(
+            functionProperties.getQueryStartClock()).getMonthValue(),
+        eval(expression).integerValue());
+    assertEquals("month_of_year(TIME '12:23:34')", expression.toString());
+  }
+
   public void testInvalidDates(String date) throws SemanticCheckException {
-    FunctionExpression expression = DSL.month_of_year(DSL.literal(new ExprDateValue(date)));
+    FunctionExpression expression = DSL.month_of_year(
+        functionProperties,
+        DSL.literal(new ExprDateValue(date)));
     eval(expression);
   }
 
   @Test void monthOfYearInvalidDates() {
```

Comment:
`@Test` on a newline and add `public`

Reply:
Fixed in 8eea6302b9a2e22afd74879d5b1b06ef684b3d64